File: src/Domain/Entities/Review.cs
using System.ComponentModel.DataAnnotations;

namespace Getir.Domain.Entities;

/// <summary>
/// Review entity for merchants and couriers
/// </summary>
public class Review
{
    public Guid Id { get; set; }
    
    [Required]
    public Guid ReviewerId { get; set; } // User who wrote the review
    
    [Required]
    public Guid RevieweeId { get; set; } // Merchant or Courier being reviewed
    
    [Required]
    public string RevieweeType { get; set; } = string.Empty; // "Merchant" or "Courier"
    
    [Required]
    public Guid OrderId { get; set; } // Order this review is for
    
    [Required]
    [Range(1, 5)]
    public int Rating { get; set; } // 1-5 stars
    
    [Required]
    [MaxLength(500)]
    public string Comment { get; set; } = string.Empty;
    
    public DateTime CreatedAt { get; set; } = DateTime.UtcNow;
    public DateTime? UpdatedAt { get; set; }
    
    // Moderation fields
    public bool IsApproved { get; set; } = true; // Auto-approve for now
    public bool IsModerated { get; set; } = false;
    public string? ModerationNotes { get; set; }
    public string? ModeratorNotes { get; set; }
    public Guid? ModeratedBy { get; set; }
    public DateTime? ModeratedAt { get; set; }
    
    // Soft delete
    public bool IsDeleted { get; set; } = false;
    public DateTime? DeletedAt { get; set; }
    
    // Navigation properties
    public virtual User Reviewer { get; set; } = null!;
    public virtual User Reviewee { get; set; } = null!;
    public virtual Order Order { get; set; } = null!;
    public virtual User? Moderator { get; set; }
    
    // Review tags/categories
    public virtual ICollection<ReviewTag> ReviewTags { get; set; } = new List<ReviewTag>();
    
    // Helpful votes
    public virtual ICollection<ReviewHelpful> ReviewHelpfuls { get; set; } = new List<ReviewHelpful>();
}

File: getir_mobile/lib/domain/entities/review.dart
import 'package:equatable/equatable.dart';

/// Review entity for merchants and couriers
class Review extends Equatable {
  final String id;
  final String reviewerId;
  final String reviewerName;
  final String revieweeId; // Merchant or Courier ID
  final String revieweeType; // "Merchant" or "Courier"
  final String orderId;
  final int rating; // 1-5 stars
  final String comment;
  final DateTime createdAt;
  final DateTime? updatedAt;
  final bool isApproved;
  final int helpfulCount; // Number of helpful votes

  const Review({
    required this.id,
    required this.reviewerId,
    required this.reviewerName,
    required this.revieweeId,
    required this.revieweeType,
    required this.orderId,
    required this.rating,
    required this.comment,
    required this.createdAt,
    this.updatedAt,
    this.isApproved = true,
    this.helpfulCount = 0,
  });

  @override
  List<Object?> get props => [
    id,
    reviewerId,
    reviewerName,
    revieweeId,
    revieweeType,
    orderId,
    rating,
    comment,
    createdAt,
    updatedAt,
    isApproved,
    helpfulCount,
  ];

  Review copyWith({
    String? id,
    String? reviewerId,
    String? reviewerName,
    String? revieweeId,
    String? revieweeType,
    String? orderId,
    int? rating,
    String? comment,
    DateTime? createdAt,
    DateTime? updatedAt,
    bool? isApproved,
    int? helpfulCount,
  }) {
    return Review(
      id: id ?? this.id,
      reviewerId: reviewerId ?? this.reviewerId,
      reviewerName: reviewerName ?? this.reviewerName,
      revieweeId: revieweeId ?? this.revieweeId,
      revieweeType: revieweeType ?? this.revieweeType,
      orderId: orderId ?? this.orderId,
      rating: rating ?? this.rating,
      comment: comment ?? this.comment,
      createdAt: createdAt ?? this.createdAt,
      updatedAt: updatedAt ?? this.updatedAt,
      isApproved: isApproved ?? this.isApproved,
      helpfulCount: helpfulCount ?? this.helpfulCount,
    );
  }
}

/// Review submission model
class SubmitReviewRequest {
  final String revieweeId;
  final String revieweeType;
  final String orderId;
  final int rating;
  final String comment;

  const SubmitReviewRequest({
    required this.revieweeId,
    required this.revieweeType,
    required this.orderId,
    required this.rating,
    required this.comment,
  });

  Map<String, dynamic> toJson() {
    return {
      'revieweeId': revieweeId,
      'revieweeType': revieweeType,
      'orderId': orderId,
      'rating': rating,
      'comment': comment,
    };
  }
}
